### DISCUSSION TITLE
Auto reload blazorise.com

### DISCUSSION THREAD

**Discussion by damianog** [Megabit/Blazorise#4348] [Ideas]
Some times I find blazorise.com page  in reaload state

![image](https://user-images.githubusercontent.com/2786682/204791798-bc7df4b1-b39c-4ee3-b272-5afdfc16b38a.png)

In similar situation I've implemented healthcheck and the script:

```javascript
<script src="_framework/blazor.server.js" autostart="false"></script>
<script>
Blazor.start({
    reconnectionHandler: {
        onConnectionDown: (options, error) => {
            var isReloading = false;

            async function attemptReload() {

                if (!isReloading) {
                    isReloading = true;
                    var request = new Request({
                        url: '/healthcheck',
                        method: 'GET'
                    });
                    var result = await fetch(request);

                    if (result.status == 200) {
                        document.location.reload();
                    }
                    isReloading = false;
                }
            }
            setInterval(attemptReload, 1500);
        },
        onConnectionUp: () => console.log("Reconnected!")
    }
});
</script>
```

**Comment by stsrki** [comment]
Thanks for the script. We will try and see if it helps us :)

**Reply by stsrki** [reply]
@damianog Is the `/healthcheck` some custom endpoint in your app? I mean, do you have a controller just for that?

**Reply by stsrki** [reply]
Like this?

```cs
using Microsoft.AspNetCore.Mvc;

namespace Blazorise.Docs.Server.Controllers;

[Route( "healthcheck" )]
[ApiController]
public class HealthcheckController : ControllerBase
{
    [HttpGet]
    public IActionResult Get()
    {
        return new OkResult();
    }
}

```

**Reply by damianog** [reply]
I just used 

```csharp
var builder = WebApplication.CreateBuilder(args);

builder.Services.AddHealthChecks();

var app = builder.Build();

app.MapHealthChecks("/healthcheck");
```

**Reply by stsrki** [reply]
> I just used
> 
> ```cs
> var builder = WebApplication.CreateBuilder(args);
> 
> builder.Services.AddHealthChecks();
> 
> var app = builder.Build();
> 
> app.MapHealthChecks("/healthcheck");
> ```

Nice. Thank you!

**Reply by David-Moreira** [reply]
Hello,
I don't think it's unnatural to go into reload, this happens when the circuit is broken, which sometimes happens when you leave a Blazor.Server tab open idly for some time.

What's the health check for here? Is that not redudant? I have some app at work with reload and I personally just try to reload each time.

**Reply by damianog** [reply]
The circuit could broke due server update or by browser or network standby, think about mobile device.
The healthcheck is only a basic point of start, if it give a response you may assume the site is UP, but you could implement a more detailed response with services state to check and force the reload only if needed.
